Tempura/UITests/UITests.swift

```diff
@@ -0,0 +1,258 @@
+   */
+  public struct ScreenSnapshot<V: ViewControllerModellableView> {
+    let viewType: V.Type
+    let models: [String: V.VM]
+    let container: Container
+    let size: CGSize
+    let hooks: [Hook: HookClosure<V>]
+    
+    /// A dictionary of configured view controllers for the various snapshot's cases
+    public var renderingViewControllers: [String: UIViewController] {
+      return self.models.mapValues { model in
+        let renderer = Renderer(self.viewType, model: model, container: self.container, size: self.size, hooks: self.hooks)
+        return renderer.getRenderingViewController()
+      }
+    }
+    
+    /// The name of the view
+    public var viewName: String {
+      return "\(self.viewType)"
+    }
+    
+    public init(
+      type: V.Type,
+      container: Container,
+      models: [String: V.VM],
+      hooks: [Hook: HookClosure<V>] = [:],
+      size: CGSize = UIScreen.main.bounds.size) {
+      
+      self.viewType = type
+      self.container = container
+      self.models = models
+      self.size = size
+      self.hooks = hooks
+    }
+  }
+  
+  
+  /// A Renderer will take a type of View, a ViewModel, a Container and will create the rendering UIViewController
+  /// that will be used to render the View.
+  /// Use `getRenderingViewController()` in order to get the configured UIViewController
+  class Renderer<V: ViewControllerModellableView> {
+    private var modellableViewType: V.Type
+    private var model: V.VM
+    private var container: Container
+    private var size: CGSize
+    private var hooks: [Hook: HookClosure<V>]
+    
+    init(_ type: V.Type, model: V.VM, container: Container, size: CGSize, hooks: [Hook: HookClosure<V>]) {
+      self.modellableViewType = type
+      self.model = model
+      self.container = container
+      self.size = size
+      self.hooks = hooks
+    }
+    
+    func getRenderingViewController() -> UIViewController {
+      
+      let containerViewController: UIViewController
+      
+      let containerVC = HookableViewController<V>()
+      containerVC.hooks = hooks
+      containerVC.rootView.model = self.model
+      
+      switch self.container {
+      case .none:
+        containerViewController = containerVC
+        
+      case .navigationController:
+        let navVC = UINavigationController(rootViewController: containerVC)
+        containerViewController = navVC
+        
+        if let hook = hooks[.navigationControllerHasBeenCreated] {
+          hook(containerVC.rootView)
+        }
+        
+      case .tabBarController:
+        let tabVC = UITabBarController()
+        tabVC.viewControllers = [containerVC]
+        containerViewController = tabVC
+      }
+      
+      containerViewController.view.frame.size = self.size
+      return containerViewController
+    }
+  }
+  
+  /// A UIViewController that allows hooks to be registered
+  class HookableViewController<V: ViewControllerModellableView>: UIViewController {
```

Comment:
Maybe you can provide an example in a test on how to use this. What do you think?

Reply:
good point, will do!